openverse_catalog/dags/common/popularity/operators.py

```diff
@@ -7,12 +7,24 @@
 logger = logging.getLogger(__name__)
 
 
+DROP_MEDIA_POPULARITY_RELATIONS_TASK_ID = "drop_media_popularity_relations"
+DROP_MEDIA_POPULARITY_FUNCTIONS_TASK_ID = "drop_media_popularity_functions"
+CREATE_MEDIA_POPULARITY_METRICS_TASK_ID = "create_media_popularity_metrics_table"
+UPDATE_MEDIA_POPULARITY_METRICS_TASK_ID = "update_media_popularity_metrics_table"
+CREATE_MEDIA_POPULARITY_PERCENTILE_TASK_ID = "create_media_popularity_percentile"
+CREATE_MEDIA_POPULARITY_CONSTANTS_TASK_ID = "create_media_popularity_constants_view"
+UPDATE_MEDIA_POPULARITY_CONSTANTS_TASK_ID = "update_media_popularity_constants_view"
+CREATE_MEDIA_STANDARDIZED_POPULARITY_TASK_ID = "create_media_standardized_popularity"
+CREATE_DB_VIEW_TASK_ID = "create_materialized_popularity_view"
+UPDATE_DB_VIEW_TASK_ID = "update_materialized_popularity_view"
+
+
 def drop_media_popularity_relations(
     postgres_conn_id,
     media_type="image",
 ):
     return PythonOperator(
-        task_id="drop_media_popularity_relations",
+        task_id=DROP_MEDIA_POPULARITY_RELATIONS_TASK_ID,
         python_callable=sql.drop_media_popularity_relations,
         op_args=[postgres_conn_id, media_type],
     )
```

Comment:
I love this change! Almost all of the functions updated in this file are pretty simple and return `PythonOperator`s. Do you think it'd make sense to move these out into the DAG proper in this PR? We can always do that in a follow up.

Reply:
Hm I thought about this, and definitely want to do this eventually. The reason I hesitated is because a lot of these functions are used in the `recreate_*_popularity_calculation` DAGs (the ones that drop all the tables/views/functions and recreate them entirely), and those are still separate.

I actually was thinking about trying to pull those DAGs into this one as well (in which case there's definitely no need for this file). I did a bunch of work to make it possible to skip the refresh popularity steps entirely with a config option, that looked like this:
<img width="1602" alt="Screen Shot 2022-05-04 at 5 44 14 PM" src="https://user-images.githubusercontent.com/63313398/167201928-c5715db2-d204-447e-a8bb-29a4effb9bb9.png">

I scrapped it (although I still have the code on a branch) because it's not useful for the majority of cases. But that use case is a valid one -- where we have SQL changes to the popularity function for example, so we need to drop and recreate all tables/functions. In that case it's unnecessary to refresh popularity data before the data refresh.

So, maybe we could do a version of what I have above, but instead of a configuration option to simply skip the refresh steps, we add a configuration option to drop and recreate tables. When enabled, we do the work currently defined in `recreate_*_popularity_calculations`, and then proceed to data refresh. If not, we go through the regular popularity flow (including the month check).

I hesitated here to over-complicate things, especially considering that recreate ought to be run quite infrequently, but there would be a couple advantages:
* When you add a new media type (like 3d models), you can just add a new DataRefreshType and generate every single popularity/data refresh DAG at once 🎉
* We save time because we don't need to refresh the matview unnecessarily if we run the data_refresh DAG after a `recreate...`

What do you think? Worthwhile as a follow-up? Or honestly, I have a lot of the work done in a local branch and could add it here 😄

Reply:
Actually it looks like `update_media_popularity_metrics` is the only one that's actually shared at the moment, so maybe we can go ahead with that part of the work at least.

Reply:
I think this is already a really solid change! I like some of the ideas you've discussed here, I think it could be a worthwhile follow up. I do worry about unnecessary complexity around the recreate steps - how often do we _really_ need to run those? If it's once a quarter, I'm not sure it makes sense to add the extra logic into this DAG. But if it's easier to maintain from a code perspective, that's a solid win...

Reply:
That was roughly my thinking, and why I ditched the original effort. The DAG gets harder to grok, especially when there are branches that can only be reached via conf.

I think as a compromise, I'll plan on a separate follow-up where I keep the `recreate_*_popularity_calculation` DAGs separate, but generate them with a factory using the `DataRefreshType` list instead of manually writing one for each model.